lib/src/platform_download.dart

```diff
@@ -0,0 +1,122 @@
+import 'dart:io';
+
+import 'package:flutter/services.dart';
+import 'package:flutter_downloader/flutter_downloader.dart';
+import 'package:flutter_downloader/src/dart_download.dart';
+
+/// A platform specific [Download] which invokes the method channel and make the
+/// download native if the platform has this requirement.
```

Comment:
I'd mention that:

- a new MethodChannel is created for every PlatformDownload
- what platforms require the downloads to be handled natively also (besides iOS and Android)? Is it even possible to download and save a file (using Flutter) without touching the native side?

Reply:
@bartekpacia I don't think iOS and Android require your downloads to be handled natively. I easily download files using the Flutter/Dart HTTP packages. I do not get smart scheduling and "real" background handling then, though.

Reply:
@lyio I agree with you for small downloads which run in the foreground, but are you sure about bigger downloads let's say more then 10 minutes in the background? Out of my understanding it is required to handle it separately on mobile platforms. I have no evidence, but I heard multiple times that you have on iOS just 10 seconds to do things in the background when your app leaves the foreground.

Reply:
@rekire I'm pretty sure there is nothing preventing you from trying to do a long download in the foreground in Flutter. But as I said, you won't get any of the benefits of background tasks. Your UI will likely jank and if you put your App in the background - the downloads will likely be stopped.

I'm not saying that any of that is a good idea. Just that I don't think the platforms are forcing you to do it properly.